src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx

```diff
@@ -161,14 +161,14 @@ function WorkspaceWorkflowsPage({policy, betas, route, reimbursementAccount, ses
                 title: translate('workflowsPage.makeOrTrackPaymentsTitle'),
                 subtitle: translate('workflowsPage.makeOrTrackPaymentsDescription'),
                 onToggle: () => {
-                    const isActive = policy?.reimbursementChoice === CONST.POLICY.REIMBURSEMENT_CHOICES.REIMBURSEMENT_YES;
-
-                    const newReimbursementChoice = isActive ? CONST.POLICY.REIMBURSEMENT_CHOICES.REIMBURSEMENT_MANUAL : CONST.POLICY.REIMBURSEMENT_CHOICES.REIMBURSEMENT_YES;
-
-                    const shouldUseDirectReimbursement = !isActive && Policy.isCurrencySupportedForDirectReimbursement(policy?.outputCurrency ?? '');
+                    const shouldUseDirectReimbursement =
+                        policy?.reimbursementChoice !== CONST.POLICY.REIMBURSEMENT_CHOICES.REIMBURSEMENT_YES &&
+                        Policy.isCurrencySupportedForDirectReimbursement(policy?.outputCurrency ?? '');
 
                     // If trying to enable direct reimbursement and currency is not supported, default to manual (indirect) reimbursement
-                    const effectiveReimbursementChoice = shouldUseDirectReimbursement ? newReimbursementChoice : CONST.POLICY.REIMBURSEMENT_CHOICES.REIMBURSEMENT_MANUAL;
+                    const effectiveReimbursementChoice = shouldUseDirectReimbursement
+                        ? CONST.POLICY.REIMBURSEMENT_CHOICES.REIMBURSEMENT_YES
+                        : CONST.POLICY.REIMBURSEMENT_CHOICES.REIMBURSEMENT_MANUAL;
```

Comment:
Almost there! If I revert the change I commented on [here](https://github.com/Expensify/App/pull/38318/commits/c1d1aa6ebc0aea1b36859d17080faf23e8f8b19a#r1532358463), I can't toggle off, because we now never set `REIMBURSEMENT_NO`. The way I see it, we need to check if we are enabling the toggle (`reimbursementChoice === REIMBURSEMENT_NO`), then switch between `REIMBURSEMENT_YES` and `REIMBURSEMENT_MANUAL` based on `shouldUseDirectReimbursement`, otherwise we are turning the toggle off so should set to `REIMBURSEMENT_NO`.

Reply:
Thank you @jjcoffee, I've pushed up a new commit [here](https://github.com/Expensify/App/pull/38318/commits/adb5d5e5f9d449bfd032849359a5443e2b249ffa), let me know what you think!